finality-aleph/src/party.rs

```diff
@@ -482,10 +494,22 @@ where
         }
     }
 
+    fn prune_session_data(&self, prune_below: SessionId) {
+        // In this method we make sure that the amount of data we keep in RAM in finality-aleph
+        // does not grow with the size of the blockchain.
+        debug!(target: "afa", "Pruning session data below {:?}.", prune_below);
+        self.session_authorities
+            .lock()
+            .retain(|&s, _| s >= prune_below);
+    }
+
     async fn run(mut self) {
         for curr_id in 0.. {
```

Comment:
Another thing that came to me – this is being run at every lunch, including one that would start from block 1000000 or whatever, so the loop below would have to go through all sessions, even if we started with the 1000000 blocks already loaded in the DB, right? Maybe we could before this point at least look in the DB we are using for the latest finalized block and use that to set the beginning of this loop? Or is my understanding somehow wrong?

Reply:
You are absolutely correct. I have added a few lines that check what's the current session (based on last finalized block) and run the iteration starting from it.